### ISSUE TITLE
Google Analytics events cap

### ISSUE THREAD

**Issue by LRitesh** [stimulant/ampm#40]
It looks like Google Analytics caps events at 500 per user session (and for physical installations it often seems to think there is only 1 session for the whole day). Wondering if there's a way to force a new session once in a while, so all events are captured?

Maybe with something like `sessionControl` here?
https://github.com/peaksandpies/universal-analytics/blob/master/AcceptableParams.md

**Comment by endquote** [comment]
Interesting. Something like this in `logging.js` seems like it might work.

```
// Log to Google Analytics.

// Restart a session every 500 events, see issue #40
this._google.eventCount = this._google.eventCount || 0;
var params = {};
if(this._google.eventCount >= 500) {
    params.sessionControl = 'start';
}

this._google.event(data.Category, data.Action, data.Label, data.Value, params);
var queue = _.clone(this._google._queue);
this._google.send(_.bind(function(error) {
    if (!error) {
        this._google.eventCount++;
        return;
    }

    if (error.code === 'ENOTFOUND') {
        // Couldn't connect -- replace the queue and try next time.
        // https://github.com/peaksandpies/universal-analytics/issues/12
        this._google._queue = queue;
    } else {
        // Something else bad happened.
        logger.warn('Error with Google Analytics', error);
    }
}, this));
```

The version of `universal-analytics` might need to be updated too... it's pretty old now.

Wanna try it out?

**Comment by endquote** [comment]
Heh, looks like I complained about this at some point myself.
https://github.com/peaksandpies/universal-analytics/issues/12#issuecomment-58433108

**Comment by LRitesh** [comment]
Haha... I was wondering why we didn't notice it before. Okay I'll try out `sessionControl` and update if that works. Thanks!

**Comment by LRitesh** [comment]
It seems to be working (I'm seeing more than 1 session over time)! I also updated universal-analytics version. Going to test it out a couple of days, then will submit a PR if all looks good.

**Comment by endquote** [comment]
Fixed in 1.7.8.
